src/main/java/cms/gov/madie/measure/services/TestCaseService.java

```diff
@@ -160,6 +164,44 @@ public List<TestCase> findTestCasesByMeasureId(String measureId) {
     return findMeasureById(measureId).getTestCases();
   }
 
+  public String deleteTestCase(String measureId, String testCaseId, String username) {
+    if (StringUtils.isBlank(testCaseId) || StringUtils.isBlank(measureId)) {
+      log.info("Test case/Measure Id cannot be null");
+      throw new InvalidIdException("Test case cannot be deleted, please contact the helpdesk");
+    }
+
+    Measure measure = findMeasureById(measureId);
+    if (!hasPermissionToDelete(username, measure)) {
+      log.info(
+          "User [{}] is not authorized to delete the test case with ID [{}] from measure [{}]",
+          username,
+          testCaseId,
+          measureId);
+      throw new UnauthorizedException("Measure", measureId, username);
+    }
+    if (measure.getTestCases() == null) {
```

Comment:
Can also check if the resulting list is empty. At least, I'm assuming it's a list.

```suggestion
    if (measure.getTestCases() == null || measure.getTestCases().isEmpty()) {
```

Reply:
sure I'll use: CollectionUtils.isEmpty(measure.getTestCases()) --> this would check for empty as well as null conditions